polygon/sync/block_downloader.go

```diff
@@ -237,6 +248,11 @@ func (d *blockDownloader) downloadBlocksUsingWaypoints(ctx context.Context, wayp
 		}
 
 		d.logger.Debug(syncLogPrefix("fetched blocks"), "len", len(blocks), "duration", time.Since(batchFetchStartTime))
+
+		sum += len(blocks)
+		for _, block := range blocks {
```

Comment:
let's move this in the goroutine that fetches the verified blocks per checkpoint - it uses a memoization technique using `waypointBlocksMemo`, so if you do this calc update after `fetchVerifiedBlocks` it you are guaranteed it will be only run once

benefit of moving this for loop inside those goroutines is that you are distributing this additional for loop for metrics across the coroutines, so less of a computational penalty here